### PR TITLE
Fix payback

### PR DESCRIPTION
**Pull request by shriyatyagii**
- Fix payback in spark and aave v3

### CODE REVIEW

File: contracts/mainnet/connectors/aave/v3/main.sol [hunk 1]
```diff
@@ -351,9 +357,17 @@ abstract contract AaveResolver is Events, Helpers {
...
+		if (_amt == uint256(-1)) {
+			uint256 _amtDSA = isEth
+				? address(this).balance
+				: tokenContract.balanceOf(address(this));
+			uint256 _amtDebt = getOnBehalfOfPaybackBalance(
+				_token,
+				rateMode,
+				onBehalfOf
+			);
+			_amt = _amtDSA <= _amtDebt ? _amtDSA : _amtDebt;
```
Comment (thrilok209) [inline contracts/mainnet/connectors/aave/v3/main.sol]:
```suggestion
			_amt = _amtDSA > _amtDebt ? _amtDebt : _amtDSA;
```

Comment (thrilok209) [reply contracts/mainnet/connectors/aave/v3/main.sol]: It's the same, but slightly cheaper

File: contracts/mainnet/connectors/spark/main.sol [hunk 2]
```diff
@@ -350,9 +357,17 @@ abstract contract SparkConnector is Events, Helpers {
...
+		if (_amt == uint256(-1)) {
+			uint256 _amtDSA = isEth
+				? address(this).balance
+				: tokenContract.balanceOf(address(this));
+			uint256 _amtDebt = getOnBehalfOfPaybackBalance(
+				_token,
+				rateMode,
+				onBehalfOf
+			);
+			_amt = _amtDSA <= _amtDebt ? _amtDSA : _amtDebt;
```
Comment (thrilok209) [inline contracts/mainnet/connectors/spark/main.sol]:
```suggestion
			_amt = amtDSA > _amtDebt ? _amtDebt : _amtDSA;
```

**Review by thrilok209 — DISMISSED** [review]

**Comment by shriyatyagii** [comment]
Update connectors ->
- `AAVE-V3-A`: [0xB1f852cF90B5641EFE20768F4caB73EA4fAC7664](https://etherscan.io/address/0xB1f852cF90B5641EFE20768F4caB73EA4fAC7664#code)
- `SPARK-A`: [0x7366A19984C06de6A91F4d33a8a8d2CaFDbA06CA](https://etherscan.io/address/0x7366A19984C06de6A91F4d33a8a8d2CaFDbA06CA#code)

Txn - 
[https://etherscan.io/tx/0xc1743229c30a48bf4aac690a7d04866439563a3c5147e6edaac4249601c1fa3b](https://etherscan.io/tx/0xc1743229c30a48bf4aac690a7d04866439563a3c5147e6edaac4249601c1fa3b)

**Review by KaymasJain — APPROVED** [review]

**Pull request merged**